docs/narr/security.rst

```diff
@@ -73,11 +70,22 @@ A simple security policy might look like the following:
 
     class SessionSecurityPolicy:
         def identify(self, request):
-            """ Return the user ID stored in the session. """
-            return request.session.get('userid')
+            """ Return app-specific user object. """
+            userid = request.session.get('userid')
+            if userid is None:
+                return None
+            return load_identity_from_db(request, userid)
+
+        def authenticated_userid(self, request):
+            """ Return a string ID for the user. """
+            identity = request.authenticated_identity
+            if identity is None:
+                return None
+            return string(identity.id)
 
-        def permits(self, request, context, identity, permission):
+        def permits(self, request, context, permission):
             """ Allow access to everything if signed in. """
+            identity = self.identify(request)
```

Comment:
When writing a security policy, we end up calling `self.identify` in all other methods.
Using `request.authenticated_identity` feels like adding a non-obvious layer of indirection when the `identify` method is right there.  But it is wasteful to process the request and fetch/create an object in every call!  So if `request.authenticated_identity` is cached, maybe this idiom (like on line 81) should be called out explicitly in the docs, with its caching benefit explained (but maybe in a later section, not here under `A simple security policy might look like the following`).

Reply:
After re-reading the ticket, I see that the caching item will be done in another discussion and PR, which seems good.  In this PR, I think the simplest thing to do is call `self.identify()` in all examples to keep things straightforward.